File: cpp/read_file.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main()
{
    int nmb;
    string line;
    ifstream myfile ("../data/numbers.txt");
    if (myfile.is_open())
    {
        while ( getline (myfile, line) )
        {
            try
            {
                nmb = stoi(line);
                cout << nmb << endl;
            }
            catch (exception e)
            {
                cout << line << " cant be parsed" << endl;
            }
        }
        myfile.close();
    }
    else 
    {
        cout << "cant open file" << endl;
    }
    return 0;
}

File: java/ReadFile.java
import java.io.*;
import java.util.*;

public class ReadFile {

    public static void main(String[] args)
    {
        ArrayList<Integer> res = new ArrayList<Integer>();
        try 
        {
            try(BufferedReader br = new BufferedReader(new FileReader("../data/numbers.txt")))
            {
                StringBuilder sb = new StringBuilder();
                String line = br.readLine();

                while(line != null)
                {
                    try
                    {
                        res.add(Integer.parseInt(line));
                    }
                    catch (NumberFormatException e)
                    {
                        System.out.println("Line " + line + " cannot be parsed");
                    }
                    line = br.readLine();
                }
            }
        }
        catch (Exception e)
        {
        }
        System.out.println(res);
    }
}
